### PR TITLE
Ignore error 2 which is an fget error at the bottom of the output we want

### PR DESCRIPTION
**Pull request by Elyrith**
This seems to fix bug #47.

### CODE REVIEW

**Comment by glensc** [comment]
i'm not confident ignoring error code 2 without knowing is that code served for such specific case.
perhaps check for "^fgets: " error and reset error code to 0?

some research what whois exit codes are, would be useful too (even if it means checking it's sourcecode)

**Comment by Elyrith** [comment]
Good point. I made an assumption without understanding the consequences and I apologize.

I did a very hasty Google search to see if I could find the error codes, but I couldn't find any reasonable results.

I did find what I think is the source code for whois (https://github.com/rfc1036/whois/tree/master), but going through that is beyond my skills at this time.

When I run test.sh (with my domain included) the output doesn't have the fget error in it because it gets printed to the terminal instead of the output file. Since the error doesn't appear in the whois output I don't think we can search for "^fgets:" to reset the error code to 0, but I'll try tonight.

**Comment by glensc** [comment]
apparently "fgets" error gets printed to stderr. need to redirect it to stdout to be greppable:

`2>&1` in https://github.com/glensc/nagios-plugin-check_domain/blob/v1.4.4/check_domain.sh#L117

`$whois ${server:+-h $server} $domain > $outfile 2>&1 && error=$? || error=$?`
or
`$whois ${server:+-h $server} $domain 2>&1 > $outfile && error=$? || error=$?`

i'm always puzzled which one is correct ;)

**Comment by glensc** [comment]
as for the sourcecode, i guess error comes from here:
https://github.com/rfc1036/whois/blob/v5.2.10/whois.c#L887

**Comment by Elyrith** [comment]
I updated my fix with your suggestions. I ran the entire thing through http://www.shellcheck.net/ to check where the 2>&1 should go and it said it needs to go last and that it is important (you can try yourself).

I also just check if that error is now in the output and, if it is, reset $error to 0.

**Comment by Elyrith** [comment]
Naturally, I can't test this now as the error is between the whois and the registrar and it's been 'fixed' by them to not show up now, but conceptually this should work and it doesn't break the script if the error doesn't occur.

File: check_domain.sh [hunk 1]
```diff
@@ -114,13 +114,16 @@ else
...
 outfile=$(tempfile)
-$whois ${server:+-h $server} $domain > $outfile && error=$? || error=$?
+$whois ${server:+-h $server} $domain > $outfile 2>&1 && error=$? || error=$?
 [ -s "$outfile" ] || die "$STATE_UNKNOWN" "UNKNOWN - Domain $domain doesn't exist or no WHOIS server available."
 
 # check for common errors
 if grep -q -e "Query rate limit exceeded. Reduced information." -e "WHOIS LIMIT EXCEEDED" $outfile; then
 	die "$STATE_UNKNOWN" "UNKNOWN - Rate limited WHOIS response"
 fi
+if grep -q -e "fgets: Connection reset by peer" $outfile; then
```
Comment (glensc) [inline check_domain.sh]: you don't need the `-e` here

**Comment by Elyrith** [comment]
Sorry that took so long to remove. I work in rotating shifts now and don't get much time during work days. :smile:

**Comment by glensc** [comment]
merged without last commit (339ab11d9feebc375562f0a5a3cb8af312f61ce6). don't know what garbage is in there. does not seem to be related

**Pull request closed without merging**